clouddriver-aws/src/test/groovy/com/netflix/spinnaker/clouddriver/aws/deploy/AutoScalingWorkerUnitSpec.groovy

```diff
@@ -1,496 +0,0 @@
-/*
- * Copyright 2014 Netflix, Inc.
- *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- *
- *    http://www.apache.org/licenses/LICENSE-2.0
- *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-
-package com.netflix.spinnaker.clouddriver.aws.deploy
-
-import com.amazonaws.services.autoscaling.AmazonAutoScaling
-import com.amazonaws.services.autoscaling.model.AlreadyExistsException
-import com.amazonaws.services.autoscaling.model.AutoScalingGroup
-import com.amazonaws.services.autoscaling.model.DescribeAutoScalingGroupsResult
-import com.amazonaws.services.ec2.AmazonEC2
-import com.amazonaws.services.ec2.model.LaunchTemplate
-import com.amazonaws.services.ec2.model.Subnet
-import com.netflix.spinnaker.config.AwsConfiguration
-import com.netflix.spinnaker.clouddriver.aws.TestCredential
-import com.netflix.spinnaker.clouddriver.aws.model.AmazonAsgLifecycleHook
-import com.netflix.spinnaker.clouddriver.aws.services.AsgService
-import com.netflix.spinnaker.clouddriver.aws.services.LaunchTemplateService
-import com.netflix.spinnaker.clouddriver.aws.services.RegionScopedProviderFactory
-import com.netflix.spinnaker.clouddriver.aws.userdata.UserDataOverride
-import com.netflix.spinnaker.clouddriver.aws.services.SecurityGroupService
-import com.netflix.spinnaker.clouddriver.data.task.DefaultTask
-import com.netflix.spinnaker.clouddriver.data.task.Task
-import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
-import com.netflix.spinnaker.clouddriver.model.Cluster
-import com.netflix.spinnaker.clouddriver.model.ClusterProvider
-import com.netflix.spinnaker.clouddriver.model.ServerGroup
-import com.netflix.spinnaker.kork.dynamicconfig.DynamicConfigService
-import org.springframework.beans.factory.annotation.Autowired
-import spock.lang.Specification
-import spock.lang.Unroll
-
-import java.time.Instant
-import java.time.temporal.ChronoUnit
-
-import static com.netflix.spinnaker.clouddriver.aws.model.AmazonAsgLifecycleHook.DefaultResult.CONTINUE
-import static com.netflix.spinnaker.clouddriver.aws.model.AmazonAsgLifecycleHook.Transition.EC2InstanceLaunching
-
-class AutoScalingWorkerUnitSpec extends Specification {
```

Comment:
It looks like a fair number of tests were removed here.  Is the equivalent functionality tested/handled elsewhere?  Can you walk me through why tests were removed here?

Reply:
The tests were removed because related source code was moved to `AsgBuilder.java`. The removed tests and more are added in test classes `AsgWithLaunchConfigurationBuilderSpec.groovy`  and `AsgWithLaunchTemplateBuilderSpec.groovy` (the diffs need to be loaded explicitly, fyi)

Reply:
Oh doh, that sneaky diff is what tricked me.  Thanks!